app/lib/blocked_relationship_map.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class BlockedRelationshipMap
```

Comment:
Perhaps the other `RelationshipMap` would simply allow you to override some of the methods with a static value?